pkg/manager/tidbngmonitoring/ng_monitoring_manager_test.go

```diff
@@ -258,33 +228,26 @@ func TestNewNGMonitorManager(t *testing.T) {
 			manager := NewNGMonitorManager(deps)
 
 			tngm := &v1alpha1.TidbNGMonitoring{}
+			tc := &v1alpha1.TidbCluster{}
 			tngm.Name = "ngm"
 			tngm.Namespace = "default"
-			if testcase.setTNGM != nil {
-				testcase.setTNGM(tngm)
+			tc.Name = "tc"
+			tc.Namespace = "default"
+			if testcase.setInputs != nil {
+				testcase.setInputs(tngm, tc)
 			}
 
 			// mock old and new sts
 			if testcase.getSts != nil {
 				old, new := testcase.getSts()
-				patch := gomonkey.ApplyFunc(GenerateNGMonitoringStatefulSet, func(tngm *v1alpha1.TidbNGMonitoring, cm *corev1.ConfigMap) (*apps.StatefulSet, error) {
+				patch := gomonkey.ApplyFunc(GenerateNGMonitoringStatefulSet, func(_ *v1alpha1.TidbNGMonitoring, _ *v1alpha1.TidbCluster, _ *corev1.ConfigMap) (*apps.StatefulSet, error) {
 					return new, nil
 				})
 				defer patch.Reset()
 				if old != nil {
 					indexer.Add(old)
 				}
 			}
-			// mock result of status update
-			populateStatusPatch := gomonkey.ApplyPrivateMethod(reflect.TypeOf(manager), "populateStatus", func(_ *ngMonitoringManager, _ *v1alpha1.TidbNGMonitoring, _ *apps.StatefulSet) error {
-				return testcase.populateStatusErr
-			})
-			defer populateStatusPatch.Reset()
-			// mock result of configmap update
-			syncConfigMapPatch := gomonkey.ApplyPrivateMethod(reflect.TypeOf(manager), "syncConfigMap", func(_ *ngMonitoringManager, _ *v1alpha1.TidbNGMonitoring, _ *apps.StatefulSet) (*corev1.ConfigMap, error) {
-				return nil, testcase.syncConfigMapErr
-			})
-			defer syncConfigMapPatch.Reset()
```

Comment:
The func `gomonkey.ApplyPrivateMethod` don't work for go1.17, so I removed two test case.